tests/providers/test_python.py

```diff
@@ -30,6 +31,40 @@ def mock_random_number(self, digits=None, fix_len=False):
         assert decimal_part == expected_decimal_part
 
 
+def test_pyfloat_right_or_left_digit_overflow():
+
+    max_float_digits = sys.float_info.dig
+    faker = Faker()
+
+    # Make random_int always return the maximum value input - makes it easy to reason about the code below
+    def mock_random_int(self, min=0, max=9999, step=1):
+        return max
+
+    # Remove the randomness from the test by mocking the `BaseProvider.random_number` value
+    def mock_random_number(self, digits=None, fix_len=False):
+        return int('12345678901234567890'[:digits or 1])
+
+    with patch('faker.providers.BaseProvider.random_int', mock_random_int):
+        with patch('faker.providers.BaseProvider.random_number', mock_random_number):
+
+            # A bit too much, but ~half on either side
+            with pytest.raises(ValueError, match='Asking for too many digits'):
+                faker.pyfloat(left_digits=max_float_digits // 2 + 1, right_digits=max_float_digits // 2 + 1)
+
+            # Asking for max digits on either side also fails, because we need one digit on the other side, i.e.
+            # 0.123123123, or 123123123.0 (at least needs to lead with `0.` or trail with `.0`).
```

Comment:
TBH - I don't think this is strictly true.. It's possible we can squeeze out one more digit, so there's that.
I honestly didn't look super close into that aspect - but I also figured not many people (if any) are working right at the edges of that their system supports w.r.t. floats.
I can be convinced to take some time and nail this down 100% tho.

Reply:
I think it's good enough for now. Thank you so much!